websites/C/Currys/presence.ts

```diff
@@ -0,0 +1,91 @@
+const presence = new Presence({
+		clientId: "996516483150663721",
+	}),
+	{ pathname, host } = document.location,
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData = {
+		smallImageKey: "search",
+		largeImageKey: "https://i.imgur.com/yOolF4f.png",
+		largeImageText: "Browsing Currys",
+		startTimestamp: browsingTimestamp,
+		details: "Browsing",
+		state: "Currys Website",
+	};
+	if (host === "www.currys.co.uk") {
+		presenceData.startTimestamp = browsingTimestamp;
+		presenceData.state = "Homepage";
+	}
+
+	// Product categories / Sub-categories
+	if (document.querySelector(".amp-herobanner-headline")) {
+		presenceData.state = document.querySelector(
+			".amp-herobanner-headline"
+		).textContent;
+	}
+	if (document.querySelector(".plp-heading"))
+		presenceData.state = document.querySelector(".plp-heading").textContent;
+
+	if (document.querySelector("#clp-row-1 > div > div > h1")) {
+		presenceData.state = document.querySelector(
+			"#clp-row-1 > div > div > h1"
+		).textContent;
+	} else if (document.querySelector("#clp-row-1 > div > h1")) {
+		presenceData.state = document.querySelector(
+			"#clp-row-1 > div > h1"
+		).textContent;
+	}
+
+	// Product page
+	if (
+		pathname.includes("/products/") &&
+		document.querySelector(".product-name")
+	)
+		presenceData.state = document.querySelector(".product-name").textContent;
```

Comment:
Change this to switch statement for better readability.

Example can be found here:
https://github.com/PreMiD/Presences/blob/0938ab7a268af83bb9c904f72ae56f927533693a/websites/C/Chessly/presence.ts#L20

Reply:
Done.